hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerApplicationAttempt.java

```diff
@@ -646,22 +646,17 @@ public int getNumReservedContainers(
   @SuppressWarnings("unchecked")
   public void containerLaunchedOnNode(ContainerId containerId,
       NodeId nodeId) {
-    writeLock.lock();
```

Comment:
Shouldn't this at least take the read lock?

Reply:
Updated the code.